### PR TITLE
Runroot: Update verify command and refactor

### PR DESCRIPTION
**Pull request by chitianhao**
This PR is a rewrite of `verify` command of `traffic_layout`.

Previously, `verify` is not very powerful because it only verifies the directory itself but not files within, and the `--fix` option only uniformly set permissions on all those files within.

With this update, `verify` command can verify the directories and files in a better way and `--fix` also is able to set the correct permission nicely.

There is also some refactoring performed.

### CODE REVIEW

**Comment by chitianhao** [comment]
Don't merge, It's WIP.

**Comment by randall** [comment]
[approve ci autest]

**Comment by randall** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
Seems reasonable, but I do think the fix and check functions should be unified, to avoid all that (almost) duplicated logic. It might be better to add an enum to the arguments of "check" or "fix". It wouldn't hurt fixing to check if it's broken first, and then if (1) broken and (2) fix enabled, fix it.

File: src/traffic_layout/engine.cc [hunk 1]
```diff
@@ -565,83 +543,99 @@ from_group(std::string const &user, gid_t group_id, gid_t path_gid)
...
 {
-  // active group and user of the path
-  std::pair<gid_t, uid_t> giduid = get_giduid(user);
+  bool new_line = false;
 
-  gid_t ts_gid = giduid.first;
-  uid_t ts_uid = giduid.second;
+  passwd *pwd = getpwnam(user.c_str());
+  if (!pwd) {
+    ink_fatal("missing password database entry for '%s'", user.c_str());
```
Comment (dragon512) [inline src/traffic_layout/engine.cc]: Are we sure this is the correct message?

Comment (chitianhao) [reply src/traffic_layout/engine.cc]: It means we can't find that username. I think this message is not bad. Any other suggestions?

**Pull request merged**